pulumi/infra/upstream_stacks.py

```diff
@@ -0,0 +1,13 @@
+import pulumi
+from infra import config
+
+class UpstreamStacks:
```

Comment:
Not sure if this is an issue yet, but you _may_ need to eventually look into making this class a singleton (if that's something that exists in Python, that is). Pulumi gets cranky if you try to create multiple stack references to the same stack in a single run.